### ISSUE TITLE
Improve the test: Validate OOM score adjustments once the node is setup pod infra containers oom-score-adj should be -998 and best effort container's should be 1000

### ISSUE THREAD

**Issue by Random-Liu** [kubernetes/kubernetes#37580]
The test `Validate OOM score adjustments once the node is setup pod infra containers oom-score-adj should be -998 and best effort container's should be 1000` is failing in `kubelet-cri-serial-gce-e2e`.
* https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-node-kubelet-cri-serial/12
* https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-node-kubelet-cri-serial/10
* https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-node-kubelet-cri-serial/4

The reason is that the test expects that all pause containers (`pidof pause`) should have `oom_score_adj=-998`. However, in many tests, we are starting test containers with `pause`. Those pause containers have `oom_score_adj=1000`.

Ideally, containers started in previous test should be cleaned up, so in serial suite it should not affect later test. However, it is not always true when CRI is enabled.

For example, in the oom message, we could see:
```
Nov 26 03:36:19 tmp-node-e2e-36829672-gci-dev-56-8977-0-0 kernel: [ pid ]   uid  tgid total_vm      rss nr_ptes nr_pmds swapents oom_score_adj name

Nov 26 03:36:19 tmp-node-e2e-36829672-gci-dev-56-8977-0-0 kernel: [ 9190]     0  9190      257        1       4       2        0          1000 pause
Nov 26 03:36:19 tmp-node-e2e-36829672-gci-dev-56-8977-0-0 kernel: [16567]     0 16567      257        1       4       2        0          1000 pause
Nov 26 03:36:19 tmp-node-e2e-36829672-gci-dev-56-8977-0-0 kernel: [16588]     0 16588      257        1       4       2        0          -998 pause
Nov 26 03:36:19 tmp-node-e2e-36829672-gci-dev-56-8977-0-0 kernel: [23573]     0 23573      257        1       3       2        0          -998 pause
Nov 26 03:36:19 tmp-node-e2e-36829672-gci-dev-56-8977-0-0 kernel: [24467]     0 24467   997933     9819     487       8        0          -999 docker
Nov 26 03:36:19 tmp-node-e2e-36829672-gci-dev-56-8977-0-0 kernel: [24482]     0 24482   104791      897      28       5        0          -999 containerd
Nov 26 03:36:19 tmp-node-e2e-36829672-gci-dev-56-8977-0-0 kernel: [24717]     0 24717    48267       85      16       5        0          -999 containerd-shim
Nov 26 03:36:19 tmp-node-e2e-36829672-gci-dev-56-8977-0-0 kernel: [24733]     0 24733      257        1       4       2        0          -998 pause
Nov 26 03:36:19 tmp-node-e2e-36829672-gci-dev-56-8977-0-0 kernel: [24768]     0 24768    48267       85      16       5        0          -999 containerd-shim
Nov 26 03:36:19 tmp-node-e2e-36829672-gci-dev-56-8977-0-0 kernel: [24783]     0 24783    20771    19149      44       5        0          -998 stress
Nov 26 03:36:19 tmp-node-e2e-36829672-gci-dev-56-8977-0-0 kernel: [24794]     0 24794    48267       84      16       5        0          -999 containerd-shim
Nov 26 03:36:19 tmp-node-e2e-36829672-gci-dev-56-8977-0-0 kernel: [24811]     0 24811      257        1       4       2        0          -998 pause
Nov 26 03:36:19 tmp-node-e2e-36829672-gci-dev-56-8977-0-0 kernel: [24843]     0 24843    48267       84      16       5        0          -999 containerd-shim
Nov 26 03:36:19 tmp-node-e2e-36829672-gci-dev-56-8977-0-0 kernel: [24857]     0 24857   328358   324704     641       6        0           973 stress
Nov 26 03:36:19 tmp-node-e2e-36829672-gci-dev-56-8977-0-0 kernel: [24868]     0 24868    48267       85      16       5        0          -999 containerd-shim
Nov 26 03:36:19 tmp-node-e2e-36829672-gci-dev-56-8977-0-0 kernel: [24883]     0 24883      257        1       4       2        0          -998 pause
Nov 26 03:36:19 tmp-node-e2e-36829672-gci-dev-56-8977-0-0 kernel: [24915]     0 24915    48267       84      16       5        0          -999 containerd-shim
Nov 26 03:36:19 tmp-node-e2e-36829672-gci-dev-56-8977-0-0 kernel: [24930]     0 24930   325187   323575     638       6        0          1000 stress
```

There are 2 problems here:
1. **Why are containers not properly cleaned up when CRI enabled?** I can't even find which test leaked pause container from the log, because there are many test using pause image, and some of them start a lot of containers, such as density test, restart test.
2. **The oom score adj test should not rely on process name `pause`**. However, I can't find an alternative for now, because we don't report pause container status in our api, we don't even know the container id to inspect with docker or CRI.

@yujuhong @vishh 
/cc @kubernetes/sig-node

**Comment by yujuhong** [comment]
This test is still flaking in cri-serial. Some examples:
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-node-kubelet-cri-serial/86
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-node-kubelet-cri-serial/84

Need to check the logs to see if the symptoms are the same.

**Comment by yujuhong** [comment]
> Why are containers not properly cleaned up when CRI enabled? I can't even find which test leaked pause container from the log, because there are many test using pause image, and some of them start a lot of containers, such as density test, restart test.

It could be that with CRI, it takes kubelet longer to clean up. Or, even worse, that kubelet is not cleaning up properly. We need more information on this on failure.
 - Dumping pods on the node 
 - Dumping containers (`docker ps`) on the node. Accessing docker directly is not ideal, but it's okay because the test itself only applies to docker. Alternatively, we can get the current running containers from kubelet's /runningPods endpoint. The downside is that this may lag behind the actual state a little bit (seconds).
We should also make the test suite run more frequently. It runs once a day, which makes the debugging round-trip overhead too high.

**Comment by yujuhong** [comment]
I can reproduce this process leak by running the "Network should recover from ip leaks" on my own node, *without* CRI enabled. The image used in the test was `gci-dev-57-9202-20-0`, and the docker version 1.11.2.
The test creates 100 pods, then restarts the docker service by `systemctl restart docker` 6 times, with 20s interval in between. After the test ends, there could be multiple processes running `/pause` left on the node. Those processes were reparented to pid 1, and `docker ps -a` showed no containers running. 

/cc @freehan

**Comment by yujuhong** [comment]
On the side note, the test should not be blindly checking all `pause` processes. It should, instead, take a snapshot of the system before the test and only check the new `pause` processes created by the tests.

**Comment by yujuhong** [comment]
I filed a new issue for the docker bug since it's not unique to CRI and this test is catching it by accident: #41450

Leaving this issue to fix the test by not blindly checking all `pause` processes: https://github.com/kubernetes/kubernetes/issues/37580#issuecomment-279569760

**Comment by Random-Liu** [comment]
@calebamiles The docker bug is not a regression, and is tracked in #41450.

The problem of the test is not necessarily to be fixed in v1.6, at least it's not release blocker. :)
